File: src/main/java/com/negosud/api/service/ArticleService.java
package com.negosud.api.service;

import com.negosud.api.dto.request.ArticleRequestDto;
import com.negosud.api.dto.request.ReapprovisionnementDto;
import com.negosud.api.dto.request.StockAjustementDto;
import com.negosud.api.dto.response.ArticleResponseDto;

import java.util.List;

public interface ArticleService {

    List<ArticleResponseDto> findAll();

    List<ArticleResponseDto> findByFamille(Long familleId);

    List<ArticleResponseDto> findByFournisseur(Long fournisseurId);

    ArticleResponseDto findById(Long id);

    ArticleResponseDto create(ArticleRequestDto dto);

    ArticleResponseDto update(Long id, ArticleRequestDto dto);

    ArticleResponseDto ajusterStock(Long id, StockAjustementDto dto);

    ArticleResponseDto toggleReapprovisionnement(Long id, ReapprovisionnementDto dto);

    void delete(Long id);
}

File: front/src/services/articleService.js
import api from './api';

const BASE = '/articles';

export const articleService = {
  findAll: () => api.get(BASE),
  findById: (id) => api.get(`${BASE}/${id}`),
  findByFamille: (familleId) => api.get(`${BASE}/famille/${familleId}`),
  findByFournisseur: (fournisseurId) => api.get(`${BASE}/fournisseur/${fournisseurId}`),
  create: (data) => api.post(BASE, data),
  update: (id, data) => api.put(`${BASE}/${id}`, data),
  ajusterStock: (id, data) => api.patch(`${BASE}/${id}/stock`, data),
  toggleReapprovisionnement: (id, data) => api.patch(`${BASE}/${id}/reapprovisionnement`, data),
  delete: (id) => api.delete(`${BASE}/${id}`),
};
